libs/app-example-auth/src/lib/user-level.guard.ts

```diff
@@ -3,28 +3,31 @@ import { AuthStateService } from './auth-state.service'
 import { UserLevel } from '@myorg/app-example-models'
 import { inject } from '@angular/core'
 
-export const userLevelGuard: CanMatchFn = (route, segments) => {
-    const authStateService = inject(AuthStateService)
-    const router = inject(Router)
 
-    const userLevel = authStateService.getUserLevel()
-    const groupId = authStateService.getGroupId()
+export function userLevelGuardFn(allowedLevels: UserLevel[]): CanMatchFn {
+    return (route, segments) => {
+        const authStateService = inject(AuthStateService)
+        const router = inject(Router)
 
-    // Allow access if:
-    // 1. User is Admin/Moderator and has a groupId, OR
-    // 2. User is User and has no groupId
-    if (
-        ((userLevel === UserLevel.Admin || userLevel === UserLevel.Moderator) &&
-            groupId) ||
-        (userLevel === UserLevel.User && groupId === '')
-    ) {
-        console.log('Access granted')
-        return true
-    }
+        const userLevel = authStateService.getUserLevel()
+        const groupId = authStateService.getGroupId()
+
+        // Allow access if:
+        // 1. User is Admin/Moderator and has a groupId, OR
+        // 2. User is User and has no groupId
+        if (
+            ((userLevel === UserLevel.Admin || userLevel === UserLevel.Moderator) &&
+                groupId) ||
+            (userLevel === UserLevel.User && groupId === '')
+        ) {
+            console.log('Access granted')
+            return true
+        }
```

Comment:
Since we are proving the levels via method argument, we will check against those.

```suggestion
        if (allowedLevels.include(userLevel)) {
            return true
        }
```